packages/ogmios/src/ogmiosToCore/tx.ts

```diff
@@ -65,319 +53,305 @@ const mapPoolParameters = (poolParameters: Schema.PoolParameters): Cardano.PoolP
+      } as Cardano.PoolRegistrationCertificate;
+    case 'stakePoolRetirement':
+      return {
+        __typename: Cardano.CertificateType.PoolRetirement,
+        epoch: Cardano.EpochNo(certificate.stakePool.retirementEpoch),
+        poolId: Cardano.PoolId(certificate.stakePool.id)
+      };
+    case 'genesisDelegation':
+      return {
+        __typename: Cardano.CertificateType.GenesisKeyDelegation,
+        genesisDelegateHash: Crypto.Hash28ByteBase16(certificate.delegate.id),
+        genesisHash: Crypto.Hash28ByteBase16(certificate.issuer.id),
+        vrfKeyHash: Crypto.Hash32ByteBase16(certificate.issuer.vrfVerificationKeyHash)
+      };
+    default:
+      throw new NotImplementedError(`Unknown certificate mapping: ${certificate}`);
   }
-  if ('stakeKeyDeregistration' in certificate) {
-    return {
-      __typename: Cardano.CertificateType.StakeKeyDeregistration,
-      stakeKeyHash: Crypto.Ed25519KeyHashHex(certificate.stakeKeyDeregistration)
-    };
-  }
-  if ('poolRegistration' in certificate) {
-    return {
-      __typename: Cardano.CertificateType.PoolRegistration,
-      poolParameters: mapPoolParameters(certificate.poolRegistration)
-    } as Cardano.PoolRegistrationCertificate;
-  }
-  if ('poolRetirement' in certificate) {
-    return {
-      __typename: Cardano.CertificateType.PoolRetirement,
-      epoch: Cardano.EpochNo(certificate.poolRetirement.retirementEpoch),
-      poolId: Cardano.PoolId(certificate.poolRetirement.poolId)
-    };
-  }
-  if ('genesisDelegation' in certificate) {
-    return {
-      __typename: Cardano.CertificateType.GenesisKeyDelegation,
-      genesisDelegateHash: Crypto.Hash28ByteBase16(certificate.genesisDelegation.delegateKeyHash),
-      genesisHash: Crypto.Hash28ByteBase16(certificate.genesisDelegation.verificationKeyHash),
-      vrfKeyHash: Crypto.Hash32ByteBase16(certificate.genesisDelegation.vrfVerificationKeyHash)
-    };
-  }
-  if ('moveInstantaneousRewards' in certificate) {
-    return {
-      __typename: Cardano.CertificateType.MIR,
-      pot:
-        certificate.moveInstantaneousRewards.pot === 'reserves'
-          ? Cardano.MirCertificatePot.Reserves
-          : Cardano.MirCertificatePot.Treasury,
-      quantity: certificate.moveInstantaneousRewards.value || 0n
-      // TODO: update MIR certificate type to support 'rewards' (multiple reward acc map to coins)
-      // This is currently not compatible with core type (missing 'rewardAccount' which doesnt exist in ogmios)
-      // rewardAccount: certificate.moveInstantaneousRewards.rewards.
-      // Add test for it too.
-    } as Cardano.MirCertificate;
-  }
-  throw new NotImplementedError('Unknown certificate mapping');
+
+  // MIR certificates are now part of the proposals as gov actions
+  // https://github.com/CardanoSolutions/ogmios/blob/master/architectural-decisions/accepted/017-api-version-6-major-rewrite.md#treasury-transfers)
+
+  // TODO: translate code below from proposal into MIR certificate
```

Comment:
TODO: MIR certificate is now part of government actions. Check if we need to extract it for backwards compatibility

Reply:
I'm guessing MIR is being deprecated?

Reply:
We need to ask how MIR certificates are mapped during chain sync

Reply:
asked here: https://github.com/CardanoSolutions/ogmios/discussions/340

Reply:
So we probably need to map `GovernanceAction<TreasuryTransfer>` differently depending on era of the block. We're not doing anything with it right now so it's ok to create a separate ticket

Reply:
What's the action here?

Reply:
It is logged as a separate task